packages/react/src/Button/Button.tsx

```diff
@@ -6,26 +6,41 @@
 import clsx from 'clsx'
 import { forwardRef } from 'react'
 import type { ButtonHTMLAttributes, ForwardedRef, PropsWithChildren } from 'react'
+import { Icon } from '../Icon'
+import type { IconProps } from '../Icon'
 
 export type ButtonProps = {
+  /** An icon to add to the button. */
+  icon?: IconProps['svg']
+  /** The position of the icon. The default is after the label. */
+  iconPosition?: 'start' | 'only'
```

Comment:
Should we disallow providing an `iconPosition` without an `icon`? If so, through a discriminative union type like e.g. `Grid` does for padding. After that, the rendering conditions below can be simplified.

Reply:
I've tried it before: https://github.com/Amsterdam/design-system/blob/8d3e9226aa1ea841e5b211d7e06c344223991fc3/packages/react/src/Button/Button.tsx

Reply:
This is not about padding, but about preventing `<Button iconPosition="start">Label</Button>`.

This should do it:

```ts
type IconButtonProps = {
  /** An icon to add to the button. */
  icon: IconProps['svg']
  /** The position of the icon. The default is after the label. */
  iconPosition?: 'start' | 'only'
}

type TextButtonProps = {
  icon?: never
  iconPosition?: never
}

export type ButtonProps = {
  /** The level of prominence. Use a primary button only once per page or section. */
  variant?: 'primary' | 'secondary' | 'tertiary'
} & PropsWithChildren<ButtonHTMLAttributes<HTMLButtonElement>> &
  (IconButtonProps | TextButtonProps)
```